### PR TITLE
Migrate Grants vouchers to ConferenceVoucher

### PR DESCRIPTION
**Pull request by marcoacierno**
## What

Converts to use the Conference voucher system instead of a custom voucher system for grants.
The old system used to create the voucher on pretix and as well and the ability to send the email which we don't have automated in the Conference voucher system yet 

## ToDo

- [x] If a ConferenceVoucher already exists, and it is CO_SPEAKER we need to upgrade it to GRANT?

### CODE REVIEW

File: backend/conferences/querysets.py [hunk 1]
```diff
@@ -8,3 +11,7 @@ def for_conference(self, conference):
 
     def for_conference_code(self, conference):
         return self.filter(**{f"{self.conference_lookup_field}__code": conference})
+
+
+class ConferenceVoucherQuerySet(ConferenceQuerySetMixin, models.QuerySet):
```
Comment (estyxx) [inline backend/conferences/querysets.py:16]: Why crating a custom class ` ConferenceVoucherQuerySet` if does not add anything? Can't you just use `ConferenceQuerySetMixin` then?

File: backend/grants/admin.py [hunk 2]
```diff
@@ -303,24 +291,47 @@ def create_grant_vouchers_on_pretix(modeladmin, request, queryset):
...
+                ConferenceVoucher(
+                    conference_id=grant.conference_id,
+                    user_id=grant.user_id,
+                    voucher_code=ConferenceVoucher.generate_code(),
+                    voucher_type=ConferenceVoucher.VoucherType.GRANT,
+                )
+            )
+            continue
+
+        if existing_voucher.voucher_type == ConferenceVoucher.VoucherType.CO_SPEAKER:
```
Comment (estyxx) [inline backend/grants/admin.py:313]: What if the type is `SPEAKER`??

File: backend/conferences/models/conference_voucher.py [hunk 3]
```diff
@@ -49,6 +50,8 @@ class VoucherType(models.TextChoices):
         help_text=_("When the email was last sent"), blank=True, null=True
     )
 
+    objects = ConferenceVoucherQuerySet().as_manager()
+
```
Comment (estyxx) [inline backend/conferences/models/conference_voucher.py:54]: Not the purpose of this PR, but why this configuration is hardcoded below?? 
if we change our mind, we need to update the code... what if conferences have different discounts? 🤔

File: backend/grants/admin.py (hunk 2 above)
Comment (marcoacierno) [reply backend/grants/admin.py:313]: If the type is speaker we already give them a 100% discount and I think it is more correct to keep their voucher marked as SPEAKER, as technically their ticket is already included in the speakers budget

File: backend/conferences/querysets.py (hunk 1 above)
Comment (marcoacierno) [reply backend/conferences/querysets.py:16]: ConferenceQuerySetMixin contains only the logic of the mixin, it isn't a `models.QuerySet`

File: backend/grants/admin.py (hunk 2 above)
Comment (estyxx) [reply backend/grants/admin.py:313]: oh I see, right!

**Review by estyxx — APPROVED** [review]
LGTM

**Pull request merged**